pkg/cmd/pr/pr.go

```diff
@@ -136,73 +144,82 @@ func (o *Options) Run() error {
-			log.Logger().Warnf("no URLs to process for rule %d", i)
+			fmt.Printf("Warning: No URLs found for rule #%d, skipping...\n", i)
+			continue
 		}
+
 		o.EnvironmentPullRequestOptions.SparseCheckoutPatterns = []string{}
 		if rule.SparseCheckout {
-			o.EnvironmentPullRequestOptions.SparseCheckoutPatterns, err = o.GetSparseCheckoutPatterns(*rule)
+			o.EnvironmentPullRequestOptions.SparseCheckoutPatterns, err = o.GetSparseCheckoutPatterns(rule)
 			if err != nil {
+				fmt.Printf("Error: Failed to get sparse checkout patterns for rule #%d, error=%v\n", i, err)
 				return err
 			}
 		}
 
 		for _, gitURL := range rule.URLs {
 			if gitURL == "" {
-				log.Logger().Warnf("missing out repository %d as it has no git URL", i)
+				fmt.Printf("Warning: Skipping empty git URL in rule #%d\n", i)
 				continue
 			}
 
-			// lets clear the branch name so we create a new one each time in a loop
 			o.BranchName = ""
-			// lets reset the base branch name each time in a loop to avoid side effects when reusing pull requests
 			o.BaseBranchName = BaseBranchName
 
 			o.Function = func() error {
 				dir := o.OutDir
-
 				for _, ch := range rule.Changes {
 					err := o.ApplyChanges(dir, gitURL, ch)
 					if err != nil {
 						return fmt.Errorf("failed to apply change: %w", err)
 					}
-
 				}
 				return nil
 			}
 
-			if rule.ReusePullRequest {
-				if len(o.Labels) == 0 {
-					return fmt.Errorf("to be able to reuse pull request you need to supply pullRequestLabels in config file or --labels")
-				}
-				o.PullRequestFilter = &environments.PullRequestFilter{Labels: []string{}}
-				for _, label := range o.Labels {
-					o.PullRequestFilter.Labels = stringhelpers.EnsureStringArrayContains(o.PullRequestFilter.Labels, label)
-				}
-				if o.AutoMerge {
-					o.PullRequestFilter.Labels = stringhelpers.EnsureStringArrayContains(o.PullRequestFilter.Labels, environments.LabelUpdatebot)
-				}
-			}
-
 			pr, err := o.EnvironmentPullRequestOptions.Create(gitURL, "", o.Labels, o.AutoMerge)
 			if err != nil {
 				return fmt.Errorf("failed to create Pull Request on repository %s: %w", gitURL, err)
 			}
 			if pr == nil {
-				log.Logger().Infof("no Pull Request created")
 				continue
 			}
 			o.AddPullRequest(pr)
+
+			if rule.AssignAuthorToPullRequests {
+				author, err := o.FindCommitAuthor(o.PipelineGitUrl, o.PipelineCommitSha)
+				if err != nil {
+					return fmt.Errorf("failed to find commit author: %w", err)
+				}
+				for _, assignee := range o.PRAssignees {
+					if assignee == author {
+						fmt.Printf("Author '%s' is already in the assignees list\n", author)
+						goto skipAuthorAppend
+					}
+				}
+				o.PRAssignees = append(o.PRAssignees, author)
+			skipAuthorAppend:
+			}
+
+			// Assign users to PR
+			if len(o.PRAssignees) > 0 {
+				err = o.AssignUserToCommit(pr, o.PRAssignees, gitURL)
+				err = nil
+				if err != nil {
+					return fmt.Errorf("failed to assign user to commit: %w", err)
+				}
+			}
 		}
 	}
```

Comment:
Same as above here with the nesting. We should refactor to pull out into specific functions.

Are the original tests for this command good?